test/cc_rules/clang/BUILD

```diff
@@ -52,7 +52,7 @@ cc_library(
 cc_shared_object(
     name = "so_test",
     srcs = ["so_test.cc"],
-    pkg_config_libs = ["python3"],
+    pkg_config_cflags = ["python3"],
```

Comment:
You might need to add something like `ldflags = "-undefined dynamic_lookup"` on OSX; according to https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/ that seems to be needed (the OSX linker is a constant source of little differences like this)

Reply:
I tried to append this linker flag on the actually build_rule, however, it doesn't seem to have worked... :(

Reply:
Can you change this one to be like the other one too please? (i.e. do the CONFIG.OS check)